ngraph/frontend/paddlepaddle/src/op/flatten_contiguous_range.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "flatten_contiguous_range.hpp"
+#include <ngraph/builder/reshape.hpp>
+#include <ngraph/opsets/opset6.hpp>
+
+namespace ngraph
+{
+    namespace frontend
+    {
+        namespace pdpd
+        {
+            namespace op
+            {
+                NamedOutputs flatten_contiguous_range(const NodeContext& node)
+                {
+                    auto x_node = node.get_ng_input("X");
+                    auto shape_of_x = std::make_shared<opset6::ShapeOf>(x_node);
+                    int dims = x_node.get_partial_shape().rank().get_length();
+                    auto start_axis = node.get_attribute<int32_t>("start_axis");
+                    auto stop_axis = node.get_attribute<int32_t>("stop_axis");
+
+                    auto axis1_begin = opset6::Constant::create(element::i64, {1}, {0});
+                    auto axis1_end = opset6::Constant::create(element::i64, {1}, {start_axis});
+                    auto axis1 = std::make_shared<opset6::StridedSlice>(shape_of_x,
+                                                                        axis1_begin,
+                                                                        axis1_end,
+                                                                        std::vector<int64_t>{0},
+                                                                        std::vector<int64_t>{0});
+                    OutputVector axes{axis1,
+                                      opset6::Constant::create(element::i64, Shape{1}, {-1.0})};
+
+                    if (stop_axis < dims - 1)
+                    {
+                        auto axis2_begin =
+                            opset6::Constant::create(element::i64, {1}, {stop_axis + 1});
+                        auto axis2_end = opset6::Constant::create(element::i64, {1}, {dims});
+                        auto axis2_node =
+                            std::make_shared<opset6::StridedSlice>(shape_of_x,
+                                                                   axis2_begin,
+                                                                   axis2_end,
+                                                                   std::vector<int64_t>{0},
+                                                                   std::vector<int64_t>{0});
+                        axes.push_back(axis2_node);
+                    }
+
+                    auto new_shape_node = std::make_shared<opset6::Concat>(axes, 0);
+                    return node.default_single_output_mapping(
```

Comment:
By setting special_zero to true you make Reshape to copy input dimension placed under the dimension where 0 is placed in the reshape_pattern input. Is this really what you wanted?

Reply:
@liubo-intel  Please have check

Reply:
@jane-intel I check the implementation, I think `special_zero = true` is we want.  We don't want a  zero dimensions. Here I paste the definition of Paddle `flatten_contiguous_range` for reference.

`A 2D tensor is reshaped input tensor. The input dimensions"
              "up to axis are flattened to the outer dimension of the output"
              "and the remaining input dimensions are flattened into the inner"
              "dimension of the output`

Reply:
Got it,  will set it to false